drivers/nrf24l01p_ng/nrf24l01p_ng_netdev.c

```diff
@@ -0,0 +1,663 @@
+        return 0;
+    }
+    uint8_t frame_len = NRF24L01P_NG_ADDR_WIDTH + pl_width;
+    /* do NOT drop frame and return exact frame size */
+    if (!buf) {
+        DEBUG_PUTS("[nrf24l01p_ng] Return exact frame length");
+        return frame_len;
+    }
+    /* drop frame, content in buf becomes invalid and return -ENOBUFS */
+    if (len < frame_len) {
+        DEBUG("[nrf24l01p_ng] Buffer too small: %u < %u, dropping frame\n",
+              len, frame_len);
+        uint8_t garbage[pl_width];
+        nrf24l01p_ng_read_rx_payload(dev, garbage, pl_width);
+        return -ENOBUFS;
+    }
+    /* get received frame */
+    uint8_t dst_addr[NRF24L01P_NG_ADDR_WIDTH];
+    if (pno == NRF24L01P_NG_P0) {
+        memcpy(dst_addr, dev->urxaddr.rxaddrpx.rx_p0, sizeof(dst_addr));
+    }
+    else {
+        memcpy(dst_addr, dev->urxaddr.rxaddrpx.rx_p1, sizeof(dst_addr));
+        if (pno > NRF24L01P_NG_P1) {
+            dst_addr[NRF24L01P_NG_ADDR_WIDTH - 1] =
+                dev->urxaddr.arxaddr.rx_addr_short[pno - 2];
+        }
+    }
+    DEBUG_PUTS("[nrf24l01p_ng] Handle received frame");
+    uint8_t *frame = (uint8_t *)buf;
+    memcpy(frame, dst_addr, sizeof(dst_addr));
+    frame += sizeof(dst_addr);
+    nrf24l01p_ng_read_rx_payload(dev, frame, pl_width);
+#if IS_USED(MODULE_NRF24L01P_NG_DIAGNOSTICS)
+    nrf24l01p_ng_diagnostics_print_frame(dev, (uint8_t *)buf, frame_len);
+#endif
+    DEBUG("[nrf24l01p_ng] Received frame length: %u\n", frame_len);
+    return (int)frame_len;
+}
+
+/**
+ * @brief   NRF24L01+ @ref netdev_driver_t::send routine
+ *
+ * @param[in] netdev        Abstract network device handle
+ * @param[in] iolist        Linked list of data to be sent, where
+ *                          the base must be the destination address
+ *
+ * @return                  Size of sent payload
+ * @retval -ENOTSUP         @p iolist had no base and no next link,
+ *                          or address was too big, or too short
+ * @retval -EAGAIN          Pending date has been sent first
+ * @retval -E2BIG           Resulting frame from iolist was too big to be sent
+ */
+static
+int _send(netdev_t *netdev, const iolist_t *iolist)
+{
+    assert(netdev && iolist);
+    if (!(iolist->iol_base) || !(iolist->iol_next)) {
+        DEBUG_PUTS("[nrf24l01p_ng] No Tx address or no payload");
+        return -ENOTSUP;
+    }
+    nrf24l01p_ng_t *dev = (nrf24l01p_ng_t *)netdev;
+    uint8_t pl_width = 0;
+    uint8_t bcast_addr[] = NRF24L01P_NG_BROADCAST_ADDR;
+    uint8_t payload[NRF24L01P_NG_MAX_PAYLOAD_WIDTH];
+    nrf24l01p_ng_acquire(dev);
+    uint8_t fifo_status;
+    uint8_t status =
+        nrf24l01p_ng_read_reg(dev, NRF24L01P_NG_REG_FIFO_STATUS,
+                              &fifo_status, 1);
+    if (status & NRF24L01P_NG_FLG_IRQ) {
+        DEBUG_PUTS("[nrf24l01p_ng] Handle pending IRQ, before sending new data");
+        nrf24l01p_ng_release(dev);
+        _isr(&dev->netdev);
+        return -EAGAIN;
```

Comment:
I'm afraid the upper layer doesn't deal with this just yet :disappointed:

Reply:
My suspicion has been confirmed: When I see the ping packet loss, I get

```
2020-06-01 23:33:01,704 #  [nrf24l01p_ng] Handle pending IRQ, before sending new data
2020-06-01 23:33:02,701 # [nrf24l01p_ng] Handle pending IRQ, before sending new data
2020-06-01 23:33:33,837 # [nrf24l01p_ng] Handle pending IRQ, before sending new data
```

Now you could wait for #13943 or #11263 to reach a conclusion - or you just spin here until the ongoing IRQ is handled.

I know from experience this can get ugly - just do it in a way that can be easily removed once our networking stack has the capability of handling this.

Reply:
But I handle it in `drivers/nrf24l01p_ng/gnrc_netif_nrf24l01p_ng.c`
in `static int _nrf24l01p_ng_adpt_send(gnrc_netif_t *netif, gnrc_pktsnip_t *pkt)`:
```C
    int res;
    /* pending data in FIFO? */
    while (-EAGAIN == (res = netdev->driver->send(netdev, &iolist))) {}
    gnrc_pktbuf_release(pkt);
    return res;
```
This causes `_isr()` to be executed until all interrupts are handled and a new frame could be send.
Or is this not sufficient because someone calls my `_send()` directly?

Reply:
Oh I didn't notice that!
At a glance I don't think anyone (other than `gnrc_netif_raw`, but I think that shouldn't concern us) will call the driver's `.send` directly. 

I wonder what's going on here then. 

> Does it work better if you put them a little further apart?

What distances are you talking about?

Reply:
My _send gets called:
```C
_gnrc_netif_thread()
```
 -->
```C
res = netif->ops->send(netif, msg.content.ptr);
```

```C
_nrf24l01p_ng_adpt_send()
```
 --> 
```C
while (-EAGAIN == (res = netdev->driver->send(netdev, &iolist))) {}
```

Reply:
> What distances are you talking about?

Like 2 meters or so. It seems to me, it works a little more reliable.
But I know, that cannot be the solution. It should scale with any frame size.
> I wonder what's going on here then.

Me too.